server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -397,6 +398,11 @@ public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
         connectToNode(node, (ConnectionProfile) null);
     }
 
+    // We are skipping node validation for extensibility as extensionNode and opensearchNode(LocalNode) will have different ephemeral id's
+    public void connectToNode(final DiscoveryNode node, boolean skipValidation) {
```

Comment:
I would suggest to remove `skipValidation` and introduce dedicated `connectToExtensionNode` method instead